src/main/java/ch/njol/skript/lang/function/ScriptFunction.java

```diff
@@ -61,12 +61,22 @@ public ScriptFunction(Signature<T> sign, SectionNode node) {
 			Object[] val = params[i];
 			if (parameter.single && val.length > 0) {
 				Variables.setVariable(parameter.name, val[0], event, true);
-			} else {
-				for (Object value : val) {
-					KeyedValue<?> keyedValue = (KeyedValue<?>) value;
-					Variables.setVariable(parameter.name + "::" + keyedValue.key(), keyedValue.value(), event, true);
+				continue;
+			}
+
+			int count = 0;
+			for (Object value : val) {
+				if (value instanceof KeyedValue<?> keyedValue) {
+					Variables.setVariable(parameter.name + Variable.SEPARATOR + keyedValue.key(), keyedValue.value(), event, true);
+					continue;
 				}
+
+				// backup for if the passed argument is not a keyed value.
+				// an example of this is passing `xs: integers = (1, 2)` as a parameter.
+				Variables.setVariable(parameter.name + Variable.SEPARATOR + count, value, event, true);
```

Comment:
i don't think this is necessarily safe
The keyed values could be conflicting with this if there's a mix of keyed and non-keyed values. Is that possible?